### PR TITLE
Feat-838: Internal router  for rack to communication

### PR DESCRIPTION
**Pull request by nightfury1204**
### What is the feature/fix?

https://github.com/convox/issues-private/issues/838

### Add screenshot or video (optional)

** Any screenshot or video capture using the feature **

### Does it has a breaking change?
No

### How to use/test it?

- deploy/update rack to this version: `convox rack update nahid-internal-router2`
- enable internal router: `convox rack params set internal_router="true"`, this will install a internal loadbalancer
- now deploy the the service with `internalRouter: true`
```
services:
  web:
    build: .
    port: 3000
    internalRouter: true
    environment:
      - PORT=3000
```
- now check the service url: `convox services -a <app>`
- it shouldn't be accessible through internet
### Checklist
- [ ] New coverage tests
- [ ] Unit tests passing
- [ ] E2E tests passing
- [ ] E2E downgrade/update test passing
- [ ] Documentation updated
- [ ] No warnings or errors on Deepsource/Codecov

### CODE REVIEW

File: docs/installation/production-rack/aws.md [hunk 1]
```diff
@@ -49,6 +49,7 @@ The following environment variables are required:
 | **cert_duration**        | **2160h**              | Certification renew period                                                                                     | 
 | **cidr**                 | **10.1.0.0/16**        | CIDR range for VPC                                                                                             |
 | **high_availability**    | **true**               | Setting this to "false" will create a cluster with less reduntant resources for cost optimization              |
+| **internal_router**  |     **false**        | This will install an internal loadbalancer within the vpc |
```
Comment (Twsouza) [inline docs/installation/production-rack/aws.md:52]: nitpick:

> Install an internal loadbalancer within the vpc

**Review by Twsouza — DISMISSED** [review]
lgtm

**Review by Twsouza — DISMISSED** [review]

**Review by Twsouza — APPROVED** [review]

**Review by yuriadams — APPROVED** [review]

**Comment by beastawakens** [comment]
@nightfury1204 Can I ask a question on this please?  I'm trying to use this feature but I'm failing at being able to communicate successfully to a service running behind the internal load balancer.  Looking at the ingress created for the service, it's not receiving the address of the internal LB (address is blank).  The service domain (`*.convox.cloud`) is alias'd correctly to the internal LB, but trying to curl or access that address from another instance in the same VPC isn't getting through to the service.  I believe there's a mismatch in the routing through from the LB to the `ingress-nginx-internal` to the service ingress.  
I'll provide a sample app repo later to demonstrate, but wanted to reach out for your thoughts initially.

**Comment by nightfury1204** [comment]
> @nightfury1204 Can I ask a question on this please? I'm trying to use this feature but I'm failing at being able to communicate successfully to a service running behind the internal load balancer. Looking at the ingress created for the service, it's not receiving the address of the internal LB (address is blank). The service domain (`*.convox.cloud`) is alias'd correctly to the internal LB, but trying to curl or access that address from another instance in the same VPC isn't getting through to the service. I believe there's a mismatch in the routing through from the LB to the `ingress-nginx-internal` to the service ingress. I'll provide a sample app repo later to demonstrate, but wanted to reach out for your thoughts initially.

@beastawakens thanks i will look into it

**Comment by beastawakens** [comment]
@nightfury1204 Thank you!  Example repo here: https://github.com/beastawakens/convox-internal-example
Deploy that to a Rack with the internal router turned on and try to curl the service domain for the internal service from within the VPC... 
If you inspect the ingresses via kubectl, you will see the one for internal service does not get assigned the address of the AWS internal load balancer:

```
╰─ kubectl get ingress --all-namespaces
NAMESPACE                NAME                       CLASS            HOSTS                                                                          ADDRESS                                                                         PORTS     AGE
internal-test-system     api-ing-v1                 nginx            api.xxx.convox.cloud                                              xxx-xxx.elb.us-west-2.amazonaws.com   80, 443   9h
internal-test-system     kubernetes-ing-v1          nginx            api.xxx.convox.cloud                                              xxx-xxx.elb.us-west-2.amazonaws.com   80, 443   9h
internal-test-testname   external                   nginx            external.testname.xxx.convox.cloud                                xxx-xxx.elb.us-west-2.amazonaws.com   80, 443   2m52s
internal-test-testname   internalservice-internal   nginx-internal   internalservice.testname.xxx.convox.cloud,iptest.smile.internal                                                                                   80        2m52s
```

**Comment by nightfury1204** [comment]
> @nightfury1204 Thank you! Example repo here: https://github.com/beastawakens/convox-internal-example Deploy that to a Rack with the internal router turned on and try to curl the service domain for the internal service from within the VPC... If you inspect the ingresses via kubectl, you will see the one for internal service does not get assigned the address of the AWS internal load balancer:
> 
> ```
> ╰─ kubectl get ingress --all-namespaces
> NAMESPACE                NAME                       CLASS            HOSTS                                                                          ADDRESS                                                                         PORTS     AGE
> internal-test-system     api-ing-v1                 nginx            api.xxx.convox.cloud                                              xxx-xxx.elb.us-west-2.amazonaws.com                                                                                      80, 443   9h
> internal-test-system     kubernetes-ing-v1          nginx            api.xxx.convox.cloud                                              xxx-xxx.elb.us-west-2.amazonaws.com                                                                                      80, 443   9h
> internal-test-testname   external                   nginx            external.testname.xxx.convox.cloud                                xxx-xxx.elb.us-west-2.amazonaws.com                                                                                      80, 443   2m52s
> internal-test-testname   internal-internal          nginx-internal   internal.testname.xxx.convox.cloud,iptest.smile.internal                                                                                   80        2m52s
> ```

thanks @beastawakens i will look into that

**Comment by beastawakens** [comment]
@nightfury1204 Wondering if you have any insight on this yet?

**Comment by beastawakens** [comment]
@nightfury1204 Thank you, I would appreciate it, this is a killer feature if we can get it to work!

**Comment by nightfury1204** [comment]
@beastawakens looking into it. aws changes some default settings, probably that's why it was causing the problem.

**Comment by beastawakens** [comment]
> @beastawakens looking into it. aws changes some default settings, probably that's why it was causing the problem.

@nightfury1204 thank you! I saw your commit earlier ☺️ I'll keep tabs on your progress! 🤞

**Pull request merged**